pkg/reconciler/labeler/v1/labels.go

```diff
@@ -73,30 +72,26 @@ func (c *Reconciler) syncLabels(ctx context.Context, r *v1.Route) error {
 	}
 
 	// Use a revision accessor to manipulate the revisions.
-	racc := &revision{r: c}
-	if err := deleteLabelForNotListed(ctx, r.Namespace, r.Name, racc, revisions); err != nil {
+	if err := deleteLabelForNotListed(r.Namespace, r.Name, racc, revisions); err != nil {
 		return err
 	}
 	if err := setLabelForListed(r, racc, revisions); err != nil {
 		return err
 	}
 
 	// Use a config access to manipulate the configs.
-	cacc := &configuration{r: c}
-	if err := deleteLabelForNotListed(ctx, r.Namespace, r.Name, cacc, configs); err != nil {
+	if err := deleteLabelForNotListed(r.Namespace, r.Name, cacc, configs); err != nil {
 		return err
 	}
 	return setLabelForListed(r, cacc, configs)
 }
 
-// clearLabels removes any labels for a named route from configurations and revisions.
-func (c *Reconciler) clearLabels(ctx context.Context, ns, name string) error {
-	racc := &revision{r: c}
-	if err := deleteLabelForNotListed(ctx, ns, name, racc, sets.NewString()); err != nil {
+// ClearLabels removes any labels for a named route from configurations and revisions.
+func ClearLabels(cacc *configuration, racc *revision, ns, name string) error {
```

Comment:
the same here: you're accepting private types to a public function.
While _it_ compiles, it's not the best design.
Can we use just the generic accessor here?

Reply:
For clear we can use Accessor. I've also made it a variadic to be super-generic. All we want to do here is clear for all Accessors.